src/test/run-make-fulldeps/coverage-reports-base/expected_show_coverage.panic_unwind.txt

```diff
@@ -0,0 +1,50 @@
+    1|       |#![allow(unused_assignments)]
+    2|       |// expect-exit-status-101
+    3|       |
+    4|       |fn might_panic(should_panic: bool) {
+    5|      4|    if should_panic {
+    6|      1|        println!("panicking...");
+    7|      1|        panic!("panics");
+    8|      3|    } else {
+    9|      3|        println!("Don't Panic");
+   10|      3|    }
+   11|      3|}
+   12|       |
+   13|       |fn main() -> Result<(),u8> {
+   14|      1|    let mut countdown = 10;
+   15|     10|    while countdown > 0 {
+   16|     10|        if countdown == 1 {
+   17|      0|            might_panic(true);
+   18|     10|        } else if countdown < 5 {
+   19|      3|            might_panic(false);
+   20|     15|        }
+   21|      9|        countdown -= 1;
+   22|       |    }
+   23|      0|    Ok(())
+   24|      0|}
+   25|       |
+   26|       |// Notes:
+   27|       |//   1. Compare this program and its coverage results to those of the similar tests `abort.rs` and
+   28|       |//      `try_error_result.rs`.
+   29|       |//   2. Since the `panic_unwind.rs` test is allowed to unwind, it is also allowed to execute the
+   30|       |//      normal program exit cleanup, including writing out the current values of the coverage
+   31|       |//      counters.
+   32|       |//   3. The coverage results show (interestingly) that the `panic!()` call did execute, but it does
+   33|       |//      not show coverage of the `if countdown == 1` branch in `main()` that calls
+   34|       |//      `might_panic(true)` (causing the call to `panic!()`).
+   35|       |//   4. The reason `main()`s `if countdown == 1` branch, calling `might_panic(true)`, appears
+   36|       |//      "uncovered" is, InstrumentCoverage (intentionally) treats `TerminatorKind::Call` terminators
+   37|       |//      as non-branching, because when a program executes normally, they always are. Errors handled
+   38|       |//      via the try `?` operator produce error handling branches that *are* treated as branches in
+   39|       |//      coverage results. By treating calls without try `?` operators as non-branching (assumed to
+   40|       |//      return normally and continue) the coverage graph can be simplified, producing smaller,
+   41|       |//      faster binaries, and cleaner coverage results.
+   42|       |//   5. The reason the coverage results actually show `panic!()` was called is most likely because
+   43|       |//      `panic!()` is a macro, not a simple function call, and there are other `Statement`s and/or
+   44|       |//      `Terminator`s that execute with a coverage counter before the panic and unwind occur.
+   45|       |//   6. By best practice, programs should not panic. By design, the coverage implementation will not
+   46|       |//      incur additional cost (in program size and execution time) to improve coverage results for
+   47|       |//      an event that is not supposted to happen.
```

Comment:
I don't think this is _always_ true. Of course there are uses like when invariants are violated (bugs), and `#[should_panic]` tests which you mention below. In addition, [salsa](https://github.com/salsa-rs/salsa) supports [unwinding as a cancellation mechanism](https://docs.rs/salsa/0.16.0/salsa/struct.Runtime.html#method.is_current_revision_canceled) for queries.

Reply:
Point taken. It's still my opinion that the *best* practice is not to use `panic!` as a programming model. (Even salsa appears to support both `panic` and `Result::Err` for cancelling, if I'm not mistaken. I'm curious why not just go with the Result::Err method, unless it's for legacy reasons.)

But since this may be more my opinion, I'll rephrase.

Reply:
I think they support unwinding because otherwise cancellation becomes "infectious" and something you have to handle throughout your code, even though it isn't a classic error handling case. I think rust-analyzer uses it.

Reply:
I'm updating the comment.

Reply:
fixed